experimental/framework/fluid-static/src/signalManager.ts

```diff
@@ -0,0 +1,177 @@
+     * times as well to remove all instances.
+     *
+     * TODO: specify behavior around in-flight calls/mutability
+     * @param signalName - The name of the signal
+     * @param listener - The callback signal handler
+     * @returns This ISignalManager
+     */
+    deregisterListener(signalName: string, listener: SignalListener): ISignalManager;
+
+    submitSignal(signalName: string, payload?: Jsonable);
+
+    registerBroadcastListener(signalName: string, listener: SignalListener): ISignalManager;
+
+    deregisterBroadcastListener(signalName: string, listener: SignalListener): ISignalManager;
+
+    requestBroadcast(signalName: string, payload?: Jsonable);
+}
+
+/**
+ * Duck type of something that provides the expected signalling functionality:
+ * A way to verify we can signal, a way to send a signal, and a way to listen for incoming signals
+ */
+export interface ISignaler {
+    connected: boolean;
+    on(event: "signal", listener: (message: IInboundSignalMessage, local: boolean) => void);
+    submitSignal(type: string, content: any): void;
+}
+
+/**
+ * Note: currently experimental and under development
+ *
+ * Helper class to assist common scenarios around working with signals.  SignalManager wraps an
+ * object with signalling functionality (e.g. ContainerRuntime or FluidDataStoreRuntime) and steals
+ * its powers as its own, and can then be used in place of the original signaller.  The wrapped
+ * object is not harmed in the process.
+ */
+export class SignalManager extends TypedEventEmitter<IErrorEvent> implements ISignalManager {
+    /**
+     * Local map of registered signal handlers
+     * Map<signalName, handlerFunction[]>
+     */
+    private readonly listenerMap = new Map<string, SignalListener[]>();
+
+    private readonly managerId: string | undefined;
+
+    constructor(
+        /**
+         * Object to wrap that can submit and listen to signals
+         */
+        private readonly signaler: ISignaler,
+        /**
+         * Optional id to assign to this manager that will be attached to
+         * signal names.  Useful to avoid collisions if there are multiple
+         * signal users at the Container level
+         */
+        managerId?: string,
+    ) {
+        super();
+        this.managerId = managerId ? `#${managerId}` : undefined;
+        this.signaler.on("signal", (message: IInboundSignalMessage, local: boolean) => {
+            const listeners = this.listenerMap.get(message.type);
+            const clientId = message.clientId;
+            // Only call listeners when the runtime is connected and if the signal has an
+            // identifiable sender clientId.  The listener is responsible for deciding how
+            // it wants to handle local/remote signals
+            // eslint-disable-next-line no-null/no-null
+            if (listeners !== undefined && this.signaler.connected && clientId !== null) {
+                listeners.forEach((listener) => {
```

Comment:
I'm wondering if there's much benefit in self-managing event emission (vs. just leveraging the `EventEmitter`'s implementation).  I see that this would let a user register a specific callback for a specific signal (rather than filtering on signal type) which is kind of nice, but an alternate approach might be to just create multiple `SignalManager`s per signal type grouping.

I'd say go with what you feel like for this PR but maybe something to think about, esp. as you flesh out examples that use it.